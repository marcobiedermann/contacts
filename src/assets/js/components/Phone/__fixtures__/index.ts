import { PhoneProps } from '..';

const props: PhoneProps = {
  value: '+1 234 56789',
};

export default props;
