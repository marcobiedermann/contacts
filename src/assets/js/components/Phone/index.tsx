import React, { FC } from 'react';

export interface PhoneProps {
  value: string;
}

const Phone: FC<PhoneProps> = (props) => {
  const { value } = props;

  return <a href={`tel:${value}`}>{value}</a>;
};

export default Phone;
