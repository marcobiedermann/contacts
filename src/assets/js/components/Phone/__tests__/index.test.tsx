import { render } from '@testing-library/react';
import React from 'react';
import Phone from '..';
import props from '../__fixtures__';

describe('phone component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Phone {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
