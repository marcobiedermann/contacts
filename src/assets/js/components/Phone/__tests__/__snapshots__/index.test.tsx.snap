// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phone component should render correctly 1`] = `
<DocumentFragment>
  <a
    href="tel:+1 234 56789"
  >
    +1 234 56789
  </a>
</DocumentFragment>
`;
